tesseract_kinematics/core/include/tesseract_kinematics/core/utils.h

```diff
@@ -535,12 +474,13 @@ inline bool isValid(const std::array<FloatType, 6>& qs)
  * @param dof The length of the float array
  */
 template <typename FloatType>
-inline void harmonizeTowardZero(Eigen::Ref<VectorX<FloatType>> qs)
+inline void harmonizeTowardZero(Eigen::Ref<VectorX<FloatType>> qs,
+                                const std::vector<Eigen::Index>& redundancy_capable_joints)
```

Comment:
This is a breaking change. Should we add a default value or a deprecated version that assigns it to all joints?

Reply:
It would be nice to minimize API breakage when possible. I don't think a default value would work; an empty vector would result in a no-op function and you can't programmatically generate a range of indices to match the size of the input `qs` with a default variable. You could make the previous version of the function a wrapper for the new version though:

```suggestion
inline void harmonizeTowardZero(Eigen::Ref<VectorX<FloatType>> qs)
{
  // Programmatically make a vector of the default indices (i.e. a range of all indices with std::iota, etc.)
  VectorX<FloatType> idx = ...;
  
  // Call the new version of the function
  return harmonizeTowardZero(qs, idx);
}

```

Reply:
I made the function deprecated to be removed at a later time.